File: Practica-1/Python/Estadistica.py
#estadistica: calcular el promedio y la desviacion estandar.
#@autor Natalia Ruby
#@version 1.0 22/08/2025

import math

class estadistica:
    def __init__(self, datos):
        self.datos = datos

    def promedio(self):
        return sum(self.datos) / len(self.datos)

    def desviacion(self):
        prom = self.promedio()
        suma = sum((x - prom) ** 2 for x in self.datos)
        return math.sqrt(suma / (len(self.datos) - 1))

#(TEST)estadistica: calcular el promedio y la desviacion estandar.
#@autor Natalia Ruby
#@version 1.0 22/08/2025

if __name__ == "__main__":
    numeros = list(map(float, input("Ingrese 10 numeros: ").split()))
    if len(numeros) != 10:
        print("Ingresar exactamente 10 numeros.")
    else:
        est = estadistica(numeros)
        print(f"El promedio es {est.promedio():.2f}")
        print(f"La desviacion estandar es {est.desviacion():.5f}")

File: Practica-1/java/Estadistica.java
package com.mycompany.programacion_2;
import java.util.Scanner;

//estadistica: calcular el promedio y la desviacion estandar.
//@autor Natalia Ruby
//@version 1.0 22/08/2025

class estadistica {
    private final double[] datos;

    public estadistica(double[] datos) {
        this.datos = datos;
    }

    public double promedio() {
        double suma = 0;
        for (double num : datos) {
            suma += num;
        }
        return suma / datos.length;
    }

    public double desviacion() {
        double prom = promedio();
        double suma = 0;
        for (double num : datos) {
            suma += Math.pow(num - prom, 2);
        }
        return Math.sqrt(suma / (datos.length - 1));
    }
}

//(TEST)estadistica: calcular el promedio y la desviacion estandar.
//@autor Natalia Ruby
//@version 1.0 22/08/2025

public class testestadistica {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Ingrese 10 numeros: ");
        String[] valoresStr = sc.nextLine().split(" ");
        double[] valores = new double[valoresStr.length];

        for (int i = 0; i < valoresStr.length; i++) {
            valores[i] = Double.parseDouble(valoresStr[i]);
        }

        estadistica est = new estadistica(valores);

        System.out.printf("El promedio es %.2f%n", est.promedio());
        System.out.printf("La desviación estandar es %.5f%n", est.desviacion());
    }
}
